code/game/machinery/doors/_door.dm

```diff
@@ -129,11 +122,33 @@
 		return 0
 	return 1
 
+/obj/machinery/door/proc/set_bounds()
+	if (dir == NORTH || dir == SOUTH)
+		bound_width = width * world.icon_size
+		bound_height = world.icon_size
+	else
+		bound_width = world.icon_size
+		bound_height = width * world.icon_size
+
 /obj/machinery/door/set_density(new_density)
 	. = ..()
 	if(.)
 		explosion_resistance = density ? initial(explosion_resistance) : 0
 
+/obj/machinery/door/set_dir(new_dir)
+	if(new_dir in GLOB.cornerdirs) 
+		new_dir &= ~(NORTH|EAST)
```

Comment:
I think these two lines can be omitted, as the logic below handles this case already.